test/test_aggregate_sensors.py

```diff
@@ -0,0 +1,180 @@
+
+    def test_good(
+        self, sum_sensor: SumSensor, target: SensorSet, int_children: List[Sensor[int]]
+    ) -> None:
+        """Test behaviour when all the child sensors are valid."""
+        for sensor in int_children:
+            target.add(sensor)
+        int_children[3].value = 10
+        assert sum_sensor.reading == Reading(mock.ANY, Sensor.Status.NOMINAL, 22)
+
+    def test_remove(
+        self, sum_sensor: SumSensor, target: SensorSet, int_children: List[Sensor[int]]
+    ) -> None:
+        """Removing a sensor changes the status back to FAILURE."""
+        self.test_good(sum_sensor, target, int_children)
+        target.remove(int_children[0])
+        assert sum_sensor.reading == Reading(mock.ANY, Sensor.Status.FAILURE, 19)
+
+
+class TestSyncSensor:
+    @pytest.fixture
+    def sync_sensor(self, target: SensorSet) -> SyncSensor:
+        return SyncSensor(
+            target,
+            "sync-sensor",
+            "sync sensor",
+            name_regex=re.compile("sync-child-.*"),
+            n_children=2,
+        )
+
+    @pytest.fixture
+    def children(self) -> List[Sensor[bool]]:
+        sensors = [Sensor(bool, f"sync-child-{i}", "") for i in range(2)]
+        sensors[0].set_value(True, status=Sensor.Status.NOMINAL)
+        sensors[1].set_value(False, status=Sensor.Status.ERROR)
+        return sensors
+
+    def test_initial_state(self, sync_sensor: SyncSensor) -> None:
+        # Should be unsynchronised because children are missing
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.ERROR, False)
+
+    def test_all_children(
+        self, target: SensorSet, sync_sensor: SyncSensor, children: List[Sensor[bool]]
+    ) -> None:
+        """Test with all child sensors present."""
+        for sensor in children:
+            target.add(sensor)
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.ERROR, False)
+        children[1].set_value(True)
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.NOMINAL, True)
+        children[0].set_value(False)
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.ERROR, False)
+        children[0].set_value(True, status=Sensor.Status.FAILURE)  # Value should be ignored
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.ERROR, False)
+        children[0].set_value(True)
+        assert sync_sensor.reading == Reading(mock.ANY, Sensor.Status.NOMINAL, True)
+
+
+class TestLatestSensor:
+    """Test :class:`.LatestSensor`."""
+
+    @pytest.fixture
+    def int_sensor(self, target: SensorSet) -> LatestSensor[int]:
+        return LatestSensor(
+            target, int, "int-sensor", "int sensor", "bits", name_regex=re.compile("int-child-.*")
+        )
+
+    @pytest.fixture
+    def str_sensor(self, target: SensorSet) -> LatestSensor[str]:
+        return LatestSensor(
+            target, str, "str-sensor", "str sensor", "", name_regex=re.compile("str-child-.*")
+        )
```

Comment:
Once more, as discussed just now, not sure this fixture is entirely necessary anymore. Give it a onceover if you need to before deleting.